tests/unit/dependency/test_azure.py

```diff
@@ -0,0 +1,7 @@
+from dvc.dependency.azure import AzureDependency
+from tests.unit.dependency.test_local import TestLocalDependency
+
+
+class TestAzureDependency(TestLocalDependency):
```

Comment:
@steffansluis, Are you sure it's running locally?

`AzureRemote` remote parses path via regex:
https://github.com/iterative/dvc/blob/865d355554b27be5ea49640c14556c46f7906327/dvc/remote/azure.py#L22-L27

And, on `TestLocalDependency`, the path is of the following type which fails on regex parsing:
https://github.com/iterative/dvc/blob/f5055940a2f806ccdf8be0193115641effd69416/tests/unit/dependency/test_local.py#L14

Same thing is happening on `TestAzureOutput`. Maybe, change the path that's getting passed to `AzureRemote` for tests?

Reply:
(Having done `source tests/remotes_env`).

`python -m tests tests/unit/output/test_azure.py tests/unit/dependency/test_azure.py` shows (among other things) the following output:
```
[gw1] [ 25%] PASSED tests/unit/output/test_azure.py::TestLocalOutput::test_save_missing                                                                                                       
[gw3] [ 50%] PASSED tests/unit/dependency/test_azure.py::TestLocalDependency::test_save_missing                                                                                               
[gw0] [ 75%] PASSED tests/unit/output/test_azure.py::TestAzureOutput::test_save_missing                                                                                                       
[gw2] [100%] PASSED tests/unit/dependency/test_azure.py::TestAzureDependency::test_save_missing                                                                                               
```

The tests are basic copies of the tests for S3: https://github.com/iterative/dvc/blob/f5055940a2f806ccdf8be0193115641effd69416/tests/unit/dependency/test_s3.py#L5-L7

And looking at the tests for the HDFS output (which [also uses regexes](https://github.com/iterative/dvc/blob/f5055940a2f806ccdf8be0193115641effd69416/dvc/remote/hdfs.py#L22)) this should be fine?:
https://github.com/iterative/dvc/blob/f5055940a2f806ccdf8be0193115641effd69416/tests/unit/dependency/test_hdfs.py#L5-L7